File: examples/dev/ir-spec/mod04.ml
let rec mk_unit_remote_dot'0 : unit -> Mod03.unit0 = function
  | () -> Mod03.Unit0

let rec mk_left_remote_dot'1 : 'tA -> ('tA, _) Mod03.either = function
  | v_A -> Mod03.Left v_A

let rec mk_right_remote_dot'1 : 'tB -> (_, 'tB) Mod03.either = function
  | v_B -> Mod03.Right v_B

let rec zero_remote_dot'2 : Mod03.unit0 * 'tV -> 'tV = function
  | v_U, v_Val -> ( match v_U with Mod03.Unit0 -> v_Val )

let rec zero_remote1_dot'2 : Mod03.unit0 * 'tV -> 'tV = function
  | Mod03.Unit0, v_Val -> v_Val

let rec un_pair_remote_dot'1 : ('tA, 'tB) Mod03.pair -> 'tA * 'tB = function
  | v_Pair -> ( match v_Pair with Mod03.Pair (v_A, v_B) -> (v_A, v_B) )

File: examples/dev/ir-spec/mod04.erl
-file("dev/src/mod04.erlt", 1).

-module(mod04).

-export([mk_unit_remote_dot/0,
         mk_left_remote_dot/1,
         mk_right_remote_dot/1,
         zero_remote_dot/2,
         zero_remote1_dot/2,
         un_pair_remote_dot/1,
         funs/0,
         import1/0]).

-import(mod03, [mk_unit/0]).

-spec mk_unit_remote_dot() -> mod03:unit0().

mk_unit_remote_dot() -> {969696, mod03, unit0, unit0}.

-spec mk_left_remote_dot(A) -> mod03:either(A, _).

mk_left_remote_dot(A) ->
    {969696, mod03, either, left, A}.

-spec mk_right_remote_dot(B) -> mod03:either(_, B).

mk_right_remote_dot(B) ->
    {969696, mod03, either, right, B}.

-spec zero_remote_dot(mod03:unit0(), V) -> V.

zero_remote_dot(U, Val) ->
    case U of {969696, mod03, unit0, unit0} -> Val end.

-spec zero_remote1_dot(mod03:unit0(), V) -> V.

zero_remote1_dot({969696, mod03, unit0, unit0}, Val) ->
    Val.

-spec un_pair_remote_dot(mod03:pair(A, B)) -> {A, B}.

un_pair_remote_dot(Pair) ->
    case Pair of
        {969696, mod03, pair, pair, A, B} -> {A, B}
    end.

-spec funs() -> [fun(() -> mod03:unit0())].

funs() ->
    [fun mod03:mk_unit/0,
     fun mod04:mk_unit_remote_dot/0,
     fun mk_unit_remote_dot/0,
     fun () -> {969696, mod03, unit0, unit0} end].

-spec import1() -> mod03:unit0().

import1() -> mod03:mk_unit().



